admin/class-yoast-network-settings-api.php

```diff
@@ -0,0 +1,195 @@
+	}
+
+	/**
+	 * Registers a network setting and its data.
+	 *
+	 * @param string $option_group The group the network option is part of.
+	 * @param string $option_name  The name of the network option to sanitize and save.
+	 * @param array  $args         {
+	 *     Optional. Data used to describe the network setting when registered.
+	 *
+	 *     @type callable $sanitize_callback A callback function that sanitizes the network option's value.
+	 *     @type mixed    $default           Default value when calling `get_network_option()`.
+	 * }
+	 */
+	public function register_setting( $option_group, $option_name, $args = array() ) {
+
+		$args = wp_parse_args( $args, array(
+			'group'             => $option_group,
+			'sanitize_callback' => null,
+		) );
+
+		if ( ! isset( $this->whitelist_options[ $option_group ] ) ) {
+			$this->whitelist_options[ $option_group ] = array();
+		}
+
+		$this->whitelist_options[ $option_group ][] = $option_name;
+
+		if ( ! empty( $args['sanitize_callback'] ) ) {
+			add_filter( "sanitize_option_{$option_name}", array( $this, 'filter_sanitize_option' ), 10, 2 );
+		}
+		if ( array_key_exists( 'default', $args ) ) {
+			add_filter( "default_site_option_{$option_name}", array( $this, 'filter_default_option' ), 10, 2 );
+		}
+
+		$this->registered_settings[ $option_name ] = $args;
+	}
+
+	/**
+	 * Handles a request in which plugin network options should be updated.
+	 *
+	 * This method works similar to how option updates are handled in `wp-admin/options.php` and
+	 * `wp-admin/network/settings.php`.
+	 */
+	public function handle_update_options_request() {
+		$option_group = filter_input( INPUT_POST, 'network_option_group', FILTER_SANITIZE_STRING );
+
+		check_admin_referer( "$option_group-network-options" );
+
+		if ( ! isset( $this->whitelist_options[ $option_group ] ) ) {
+			wp_die( __( 'Sorry, you are not allowed to modify unregistered network settings.', 'wordpress-seo' ), '', 403 );
+		}
+
+		foreach ( $this->whitelist_options[ $option_group ] as $option_name ) {
+			$value = isset( $_POST[ $option_name ] ) ? $_POST[ $option_name ] : null;
+			$value = wp_unslash( $value );
+
+			WPSEO_Options::update_site_option( $option_name, $value );
+		}
+
+		$settings_errors = get_settings_errors();
+		if ( empty( $settings_errors ) ) {
+			add_settings_error( $option_group, 'settings_updated', __( 'Settings Updated.', 'wordpress-seo' ), 'updated' );
+		}
+
+		// Use a regular transient here, since it is automatically cleared right after the redirect.
+		// A network transient would be cleaner, but would require a lot of copied code from core for
+		// just a minor adjustment when displaying settings errors.
+		set_transient( 'settings_errors', get_settings_errors(), 30 );
+
+		$sendback = add_query_arg( 'settings-updated', 'true', wp_get_referer() );
+		wp_safe_redirect( $sendback );
+		exit;
+	}
+
+	/**
+	 * Sanitization callback to use instead of the actual callback to the `register_setting()` method.
```

Comment:
Please rewrite this to fit the documentation standards (i.e. `Sanitizes [...]`).